x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/topn/TopNEncoderTests.java

```diff
@@ -0,0 +1,130 @@
+        int v = randomInt();
+        encoder.encodeInt(v, builder);
+        BytesRef encoded = builder.toBytesRef();
+        assertThat(encoder.decodeInt(encoded), equalTo(v));
+        assertThat(encoded.length, equalTo(0));
+    }
+
+    public void testDouble() {
+        BytesRefBuilder builder = new BytesRefBuilder();
+        double v = randomDouble();
+        encoder.encodeDouble(v, builder);
+        BytesRef encoded = builder.toBytesRef();
+        assertThat(encoder.decodeDouble(encoded), equalTo(v));
+        assertThat(encoded.length, equalTo(0));
+    }
+
+    public void testBoolean() {
+        BytesRefBuilder builder = new BytesRefBuilder();
+        boolean v = randomBoolean();
+        encoder.encodeBoolean(v, builder);
+        BytesRef encoded = builder.toBytesRef();
+        assertThat(encoder.decodeBoolean(encoded), equalTo(v));
+        assertThat(encoded.length, equalTo(0));
+    }
+
+    public void testAlpha() {
+        assumeTrue("unsupported", encoder == TopNEncoder.UTF8);
+        roundTripBytesRef(new BytesRef(randomAlphaOfLength(6)));
+    }
+
+    public void testUtf8() {
+        assumeTrue("unsupported", encoder == TopNEncoder.UTF8);
+        roundTripBytesRef(new BytesRef(randomRealisticUnicodeOfLength(6)));
+    }
+
+    /**
+     * Round trip the highest unicode character to encode without a continuation.
+     */
+    public void testDel() {
+        assumeTrue("unsupported", encoder == TopNEncoder.UTF8);
+        roundTripBytesRef(new BytesRef("\u007F"));
+    }
+
+    /**
+     * Round trip the lowest unicode character to encode using a continuation byte.
+     */
+    public void testPaddingCharacter() {
+        assumeTrue("unsupported", encoder == TopNEncoder.UTF8);
+        roundTripBytesRef(new BytesRef("\u0080"));
+    }
+
+    public void testVersion() {
+        assumeTrue("unsupported", encoder == TopNEncoder.VERSION);
+        roundTripBytesRef(randomVersion().toBytesRef());
+    }
+
+    public void testIp() {
+        assumeTrue("unsupported", encoder == TopNEncoder.IP);
+        roundTripBytesRef(new BytesRef(InetAddressPoint.encode(randomIp(randomBoolean()))));
+    }
+
+    private void roundTripBytesRef(BytesRef v) {
+        BytesRefBuilder builder = new BytesRefBuilder();
+        int reportedSize = encoder.encodeBytesRef(v, builder);
+        BytesRef encoded = builder.toBytesRef();
+        assertThat(encoded.length, equalTo(reportedSize));
+        assertThat(encoder.decodeBytesRef(encoded, new BytesRef()), equalTo(v));
+        assertThat(encoded.length, equalTo(0));
+    }
+
+    static Version randomVersion() {
+        // TODO degenerate versions and stuff
+        return switch (between(0, 2)) {
+            case 0 -> new Version(Integer.toString(between(0, 100)));
+            case 1 -> new Version(between(0, 100) + "." + between(0, 100));
+            case 2 -> new Version(between(0, 100) + "." + between(0, 100) + "." + between(0, 100));
```

Comment:
Maybe add something like `case 3 -> TopNOperatorTests.randomVersion();`? Those are not randomized tests, but they are a bit more diverse.

Reply:
Turns out that's super hard! Because these are in two different projects. Annoying!

Reply:
Wait. That shouldn't be true. Checking!

Reply:
I have no idea. But it seems to work locally now. :+1: